src/main/java/com/google/devtools/build/lib/rules/cpp/CcModule.java

```diff
@@ -1690,7 +1693,12 @@ static CcToolchainFeatures.Tool toolFromStarlark(StarlarkInfo toolStruct) throws
         throw infoError(toolStruct, "\"tool\" and \"path\" cannot be set at the same time.");
       }
 
-      toolPath = PathFragment.create(toolPathString);
+      try {
+        toolPath = PathFragment.createForOs(toolPathString, execOs);
```

Comment:
Do  understand correctly that this will raise an error where there was none, but that's OK because if Bazel encountered ` C:\WINDOWS32\SYSTEM\` style path on Linux, the situation would be unsalvageable anyway?

Actually: what about the case where Bazel runs on Linux, there is a toolchain definition for Windows somewhere but it's not used? Then it looks like Bazel would raise an error where there were none. I'm not sure if this scenario happens a lot because it would require someone to depend on the toolchain, which they don't if it's just a toolchain that's never selected.